iscript/src/iscript/autograph.py

```diff
@@ -472,3 +488,59 @@ def _get_omnija_signing_files(file_list):
         sig = widevine.generate_widevine_signature(signature, certificate, flags)
         fout.write(sig)
     return to
+
+
+def langpack_id(app):
+    """Return a list of id's for the langpacks.
+
+    Side Affect of checking if filenames are actually langpacks.
+    """
+    _, file_extension = os.path.splitext(app.orig_path)
+    if not file_extension == ".xpi":
+        raise IScriptError(f"Expected an xpi got {app.orig_path}")
+
+    langpack = zipfile.ZipFile(app.orig_path, "r")
+    id = None
+    with langpack.open("manifest.json", "r") as f:
+        manifest = json.load(f)
+        if not (
+            "languages" in manifest
+            and "langpack_id" in manifest
+            and "applications" in manifest
+            and "gecko" in manifest["applications"]
+            and "id" in manifest["applications"]["gecko"]
+            and LANGPACK_RE.match(manifest["applications"]["gecko"]["id"])
```

Comment:
We may want to either use `.get()` or catch `KeyError` in these checks, in case the schema changes. Not the worst thing though, as we'll still die, just without a nice error message.

Reply:
This could be worth a followup, but I'm leaving it for now (at least because this matches the `signingscript` code)

Reply:
Filed https://github.com/mozilla-releng/scriptworker-scripts/issues/12